### DISCUSSION TITLE
Possible websocket timeout: solutions?

### DISCUSSION THREAD

**Discussion by ghost** [NginxProxyManager/nginx-proxy-manager#1661] [Q&A]
So, the web application I am proxying has been getting problems like some of the functions in the UI getting unresponsive. 

Example: when I click on the "save settings" button on the UI, it doesn't work if I kept the app open in my browser for more than ~10 minues. I have to reload the page in order to fix that. 

This doesn't happen when I access the app directly through ip:port via http. So it's likely that the problem lies in the NGINX proxy config where websocket connection between client and the server is being getting disconnected.

I have tried following solutions to no avail.
1. Turning on websocket support in NPM obviously

3. Switching to DNS mode in Cloudflare settings instead of using proxy mode because Cloudflare has a standard websocket timeout period of 100s

3. Adding keep alive timeout directive to NPM proxy config of the app. (/data/nginx/proxy_hosts/3.conf)

Also, I have noticed when I access the app via http without going through NPM(IP:Port), the browser uses the ws protocol whereas when I acces it via https through my domain, it uses the wss. Not sure if that has anything to do with my problem though (Is it because I enabled http/2 in NPM?).

I have ran out of solutions or to diagnose where the problem lies in. I'd appreciate any help on this. Thanks in advance.

**Comment by chaptergy** [comment]
Use the advanced config to increase the timeout in nginx.

```nginx
proxy_send_timeout 60m;  
proxy_read_timeout 60m;
```

The example above increases the timeout to 60 minutes. Use other values if you need more.
